TODO.md

```diff
@@ -0,0 +1,71 @@
+# TODO plan for 0.8
+
+These may not all be implemented, but are a tentatively roadmap for things to work for in 0.8.  0.8
+is really meant to be a refinement release rather than institute major underlying changes, with the
+investment paying off by cleaning up the code to allow fast and more reliable future features.
+
+## Installer build improvements
+
+Better installers so we can "make windows" or "make macos" and out pops an installer.  Currently
+installers are too manual and we need to automate this.  More specifically:
+- CPack installer for windows (32 and 64-bit).  If possible figure out how to do a "single"
+  installer that can install the right one.
+- macOS build automation (i.e. up to code signing)
+- lokinet GUI package for linux (debs)
+
+## Clean up cruft.
+
+There is some unused (non-code) junk that has built up in the repo: (.exe files, random patches,
+etc.).
+
+
+## Make `make format` less annoying
+
+New `make format' rules.  Relax some things, decide on other sane rules that are helpful without
+being annoyingly rigid.  Currently make format rules are far too rigid to the point of being
+annoying.
+
+## .ini file overhaul
+
+Right now we have a hard-coded generated .ini file that only includes some settings.  We need to
+overhaul this to include everything -- some options with initial settings, other options commented
+out -- along with descriptions of each option.  We should do this "right": i.e. by having one single
+set of specification of options that can do both parsing and config file generation rather than
+duplicating options in parsing and options printed in the generated config file (because that will
+inevitably diverge).
```

Comment:
If we're doing anything more than just generating commented-out lines for the default/example config, I'd like to consider removing some of the side effects in the config parsing code (notably, parsing the conf will change the log level).